### PR TITLE
libckteec/libteec: move LFLAGS to end of call

### PR DESCRIPTION
**Pull request by Emantor**
At least Yocto and ptxdist provide the option to invoke the linker with
-Wl,--as-needed, which will only mark the libraries as needed if they
are used by the binary or library. However this assumes that the object
files are passed before the shared libraries, move the LFLAGS for
libctkteec and libteec to the end of linker invocation to support this.

Signed-off-by: Rouven Czerwinski <rouven@czerwinskis.de>

### CODE REVIEW

File: libteec/Makefile [hunk 1]
```diff
@@ -48,7 +48,7 @@ libteec: $(TEEC_LIBRARY) $(OUT_DIR)/libteec.a
 
 $(TEEC_LIBRARY): $(TEEC_OBJS)
 	@echo "  LINK    $@"
-	$(VPREFIX)$(CC) -shared -Wl,-soname,$(LIB_MAJOR) $(TEEC_LFLAGS) -o $@ $+
+	$(VPREFIX)$(CC) -shared -Wl,-soname,$(LIB_MAJOR) -o $@ $+ $(TEEC_LFLAGS)
```
Comment (jforissier) [inline libteec/Makefile]: I don't understand why this changes anything wrt. `-Wl,--as-needed`. As long as the flag is given before the shared libraries that should be pulled as needed, the result should be the same, no?

Comment (Emantor) [reply libteec/Makefile]: At least during my testing the library is linked correctly, but is not dynamically linked to the required library. This results i.e. in libckteec not being linked to libteec, which results in an error on execution (since the TEEC_* symbols can't be resolved). 
I'm still searching for the proper LD documentation, however all I can currently come up with are stackoverflow posts and these mention that the objects need to be before the shared library flags to properly detect that the shared library is needed, [1](https://stackoverflow.com/questions/7826448/linking-libraries-with-gcc-order-of-arguments), [2](https://stackoverflow.com/questions/8207464/linking-math-library-in-gcc-4-6-1-ubuntu-11-10).

Comment (Emantor) [reply libteec/Makefile]: The [llvm-ld](https://linux.die.net/man/1/llvm-ld) man page mentions the same under library linkage.

Comment (jenswi-linaro) [reply libteec/Makefile]: Sometimes we use `--start-group` and `--end-group` to work around this. In this case it's clear that the libraries should be given after the object files.

Comment (jforissier) [reply libteec/Makefile]: Got it now. Indeed objects should always be before libraries. Now I understand why adding `-Wl,--as-needed` triggers the issue -- without it the DT_NEEDED dependencies are always added.

**Comment by Emantor** [comment]
Tag applied.

**Comment by Emantor** [comment]
This should also fix #178, added as Fixes to the commit.

**Comment by jforissier** [comment]
> This should also fix #178, added as Fixes to the commit.

@veriktig can you confirm?

**Comment by Emantor** [comment]
On second look at #178 this looks more like a problem with `-L…` and his underlying build system. I removed the fixes flag.

**Comment by jforissier** [comment]
> On second look at #178 this looks more like a problem with `-L…` and his underlying build system. I removed the fixes flag.

Agreed :wink:
